python/ray/train/torch/train_loop_utils.py

```diff
@@ -559,7 +559,17 @@ def get_device(self) -> torch.device:
         """
         if torch.cuda.is_available():
             # GPU IDs are assigned by Ray after you specify "use_gpu"
-            gpu_ids = ray.get_gpu_ids()
+            # GPU `ray.get_gpu_ids()` may return ints or may return strings.
+            # We should always convert to ints.
+            try:
+                gpu_ids = [int(id) for id in ray.get_gpu_ids()]
```

Comment:
Should we just always use string here and similarly when parsing `CUDA_VISIBLE_DEVICES` in line 584?

Reply:
Good point, updated